### ISSUE TITLE
f4m downloader producing broken video files on daisuki.net

### ISSUE THREAD

**Issue by fb39ca4** [rg3/youtube-dl#6175]
I have created an extractor for daisuki.net, https://github.com/fb39ca4/youtube-dl/blob/daisuki/youtube_dl/extractor/daisuki.py

It uses _extract_f4m_formats, and seems to complete the download fine. However, the resulting files are unplayable. Here is the output from mpv. Perhaps the stream is encrypted, and youtube-dl's implementation does not handle that?

```
[ffmpeg/demuxer] flv: Audio codec (f) is not implemented. Update your FFmpeg ver
sion to the newest one from Git. If the problem still occurs, it means that your
 file has a feature which has not been implemented.
[ffmpeg/demuxer] flv: If you want to help, upload a sample of this file to ftp:/
/upload.ffmpeg.org/incoming/ and contact the ffmpeg-devel mailing list. (ffmpeg-
devel@ffmpeg.org)
[ffmpeg/demuxer] flv: Stream #0: not enough frames to estimate rate; consider in
creasing probesize
[ffmpeg/demuxer] flv: Could not find codec parameters for stream 0 (Video: none,
 none, 848 kb/s): unknown codec
[ffmpeg/demuxer] Consider increasing the value for the 'analyzeduration' and 'pr
obesize' options
[ffmpeg/demuxer] flv: Could not find codec parameters for stream 1 (Audio: none
([15][0][0][0] / 0x000F), 11025 Hz, mono, 128 kb/s): unknown codec
[ffmpeg/demuxer] Consider increasing the value for the 'analyzeduration' and 'pr
obesize' options
[lavf] av_find_stream_info() failed
[ffmpeg/demuxer] flv: Audio codec (f) is not implemented. Update your FFmpeg ver
sion to the newest one from Git. If the problem still occurs, it means that your
 file has a feature which has not been implemented.
[ffmpeg/demuxer] flv: If you want to help, upload a sample of this file to ftp:/
/upload.ffmpeg.org/incoming/ and contact the ffmpeg-devel mailing list. (ffmpeg-
devel@ffmpeg.org)
[ffmpeg/demuxer] flv: Stream #0: not enough frames to estimate rate; consider in
creasing probesize
[ffmpeg/demuxer] flv: Could not find codec parameters for stream 0 (Video: none,
 none, 848 kb/s): unknown codec
[ffmpeg/demuxer] Consider increasing the value for the 'analyzeduration' and 'pr
obesize' options
[ffmpeg/demuxer] flv: Could not find codec parameters for stream 1 (Audio: none
([15][0][0][0] / 0x000F), 11025 Hz, mono, 128 kb/s): unknown codec
[ffmpeg/demuxer] Consider increasing the value for the 'analyzeduration' and 'pr
obesize' options
[lavf] av_find_stream_info() failed
Failed to recognize file format.
```

**Comment by dstftw** [comment]
It's DRM protected.

**Comment by fb39ca4** [comment]
Here is an example of the manifest. Do you know which part gives information about the DRM?

    <?xml version="1.0" encoding="UTF-8"?>
    <manifest xmlns="http://ns.adobe.com/f4m/1.0" xmlns:akamai="uri:akamai.com/f4m/1.0">
      <akamai:version>2.0</akamai:version>
      <akamai:bw>5000</akamai:bw>
      <id>/43383936/35470338/smil/US/00007/182865319659405.smil_0</id>
      <streamType>recorded</streamType>
      <akamai:streamType>vod</akamai:streamType>
      <duration>1450.048</duration>
      <streamBaseTime>0.000</streamBaseTime>
      <pv-2.0>;hdntl=exp=1436397406~acl=%2f*~data=hdntl~hmac=fa86dca6f0e1ca4ccb95d5d5d0f2e112b07ae47ffdfb7dc73ab9a3d241574b30</pv-2.0>
      <bootstrapInfo profile="named" id="bootstrap_1000000">AAAAi2Fic3QAAAAAAAAAAQAAAAPoAAAAAAAWIEAAAAAAAAAAAAAAAAAAAQAAABlhc3J0AAAAAAAAAAABAAAAAQAAAPEBAAAARmFmcnQAAAAAAAAD6AAAAAADAAAAAQAAAAAAAAAAAAAXcAAAAPEAAAAAABX5AAAAJ0AAAAAAAAAAAAAAAAAAAAAAAA==</bootstrapInfo>
      <bootstrapInfo profile="named" id="bootstrap_2000000">AAAAi2Fic3QAAAAAAAAAAQAAAAPoAAAAAAAWIEAAAAAAAAAAAAAAAAAAAQAAABlhc3J0AAAAAAAAAAABAAAAAQAAAPEBAAAARmFmcnQAAAAAAAAD6AAAAAADAAAAAQAAAAAAAAAAAAAXcAAAAPEAAAAAABX5AAAAJ0AAAAAAAAAAAAAAAAAAAAAAAA==</bootstrapInfo>
      <bootstrapInfo profile="named" id="bootstrap_4000000">AAAAi2Fic3QAAAAAAAAAAQAAAAPoAAAAAAAWIEAAAAAAAAAAAAAAAAAAAQAAABlhc3J0AAAAAAAAAAABAAAAAQAAAPEBAAAARmFmcnQAAAAAAAAD6AAAAAADAAAAAQAAAAAAAAAAAAAXcAAAAPEAAAAAABX5AAAAJ0AAAAAAAAAAAAAAAAAAAAAAAA==</bootstrapInfo>
      <bootstrapInfo profile="named" id="bootstrap_6000000">AAAAi2Fic3QAAAAAAAAAAQAAAAPoAAAAAAAWIEAAAAAAAAAAAAAAAAAAAQAAABlhc3J0AAAAAAAAAAABAAAAAQAAAPEBAAAARmFmcnQAAAAAAAAD6AAAAAADAAAAAQAAAAAAAAAAAAAXcAAAAPEAAAAAABX5AAAAJ0AAAAAAAAAAAAAAAAAAAAAAAA==</bootstrapInfo>
      <media bitrate="996" url="1000000_456456fdb7541131_" bootstrapInfoId="bootstrap_1000000">
        <metadata>AgAKb25NZXRhRGF0YQgAAAAMAAhkdXJhdGlvbgBAlqgxJul41QAFd2lkdGgAQIQAAAAAAAAABmhlaWdodABAdoAAAAAAAAANdmlkZW9kYXRhcmF0ZQBAiz77yRZQvgAJZnJhbWVyYXRlAEA3+ctHdzzYAAx2aWRlb2NvZGVjaWQAQBwAAAAAAAAADWF1ZGlvZGF0YXJhdGUAQF9YAAAAAAAAD2F1ZGlvc2FtcGxlcmF0ZQBA53AAAAAAAAAPYXVkaW9zYW1wbGVzaXplAEAwAAAAAAAAAAZzdGVyZW8BAQAMYXVkaW9jb2RlY2lkAEAkAAAAAAAAAAhmaWxlc2l6ZQBBpYxFagAAAAAACQ==</metadata>
      </media>
      <media bitrate="1995" url="2000000_456456fdb7541131_" bootstrapInfoId="bootstrap_2000000">
        <metadata>AgAKb25NZXRhRGF0YQgAAAAMAAhkdXJhdGlvbgBAlqgxJul41QAFd2lkdGgAQIqgAAAAAAAABmhlaWdodABAfgAAAAAAAAANdmlkZW9kYXRhcmF0ZQBAnTvaDXIbowAJZnJhbWVyYXRlAEA3+ctHdzzYAAx2aWRlb2NvZGVjaWQAQBwAAAAAAAAADWF1ZGlvZGF0YXJhdGUAQF9YAAAAAAAAD2F1ZGlvc2FtcGxlcmF0ZQBA53AAAAAAAAAPYXVkaW9zYW1wbGVzaXplAEAwAAAAAAAAAAZzdGVyZW8BAQAMYXVkaW9jb2RlY2lkAEAkAAAAAAAAAAhmaWxlc2l6ZQBBtZFdpgAAAAAACQ==</metadata>
      </media>
      <media bitrate="3368" url="4000000_456456fdb7541131_" bootstrapInfoId="bootstrap_4000000">
        <metadata>AgAKb25NZXRhRGF0YQgAAAAMAAhkdXJhdGlvbgBAlqgxJul41QAFd2lkdGgAQJQAAAAAAAAABmhlaWdodABAhoAAAAAAAAANdmlkZW9kYXRhcmF0ZQBAqNfm0I5GtQAJZnJhbWVyYXRlAEA3+ctHdzzYAAx2aWRlb2NvZGVjaWQAQBwAAAAAAAAADWF1ZGlvZGF0YXJhdGUAQGesAAAAAAAAD2F1ZGlvc2FtcGxlcmF0ZQBA53AAAAAAAAAPYXVkaW9zYW1wbGVzaXplAEAwAAAAAAAAAAZzdGVyZW8BAQAMYXVkaW9jb2RlY2lkAEAkAAAAAAAAAAhmaWxlc2l6ZQBBwjNZ+4AAAAAACQ==</metadata>
      </media>
      <media bitrate="5195" url="6000000_456456fdb7541131_" bootstrapInfoId="bootstrap_6000000">
        <metadata>AgAKb25NZXRhRGF0YQgAAAAMAAhkdXJhdGlvbgBAlqgxJul41QAFd2lkdGgAQJ4AAAAAAAAABmhlaWdodABAkOAAAAAAAAANdmlkZW9kYXRhcmF0ZQBAs45Lhae1ygAJZnJhbWVyYXRlAEA3+ctHdzzYAAx2aWRlb2NvZGVjaWQAQBwAAAAAAAAADWF1ZGlvZGF0YXJhdGUAQGesAAAAAAAAD2F1ZGlvc2FtcGxlcmF0ZQBA53AAAAAAAAAPYXVkaW9zYW1wbGVzaXplAEAwAAAAAAAAAAZzdGVyZW8BAQAMYXVkaW9jb2RlY2lkAEAkAAAAAAAAAAhmaWxlc2l6ZQBBzBD0XYAAAAAACQ==</metadata>
      </media>
    </manifest>

**Comment by fstirlitz** [comment]
This isn't the usual FlashAccess DRM which puts tags in the manifest; it seems to be the custom Akamai DRM. The only piece of information available on it seems to be [a forum post by the author of the AdobeHDS.php script](http://stream-recorder.com/forum/showpost.php?p=59493&postcount=40), and it may be outdated.

**Comment by fb39ca4** [comment]
Ok. I'm instead going to investigate using HLS for daisuki, but I need to figure out how to get youtube-dl to use the proper cookies when downloading m3u8 files.

**Comment by yan12125** [comment]
Daisuki has moved to HLS and I've implemented it in 9bd05b5.
